### DISCUSSION TITLE
Bot goes offline after 10-20secs

### DISCUSSION THREAD

**Discussion by developers192** [Rapptz/discord.py#7204] [Q&A] [answered]
Hello. I'm a new user just started writing bots with discord.py. Recently I've written a ModMail bot. When i try running the bot, it does get online and responds to commands, as expected. But after 10-20 seconds, the bot goes offline but the python file in my pc still running, no errors in console. Can anyone help me with this?
My bot's code: https://pastebin.com/SDWjf8Rb (Token removed)
Sorry for my messy code arrangements.

EDIT: Here's my bot log
[bot.log](https://github.com/Rapptz/discord.py/files/6796560/bot.log)

**Comment by lmaotrigine** [comment, marked as answer]
You have the event loop blocked because of your use of the `with` context manager without yielding from it. 

There will also be issues because you define and start tasks inside of `on_ready` and changing presence every 4 seconds. 

As for your immediate issue, consider using a database to store data persistently.

**Reply by developers192** [reply]
Thanks for your reply, I will look into it.

**Reply by Vexs** [reply]
The existence of a `with` does not mean the code is being blocked. There are, presumably, some non-async context managers that can block but I see none here.

**Reply by Phxntxm** [reply]
The first time a message is sent a `while True` loop is started, and reads a line from a file... then does things based on that. If those ifs do not pass then it will permanently block since those are the only asynchronous things I see in this while loop. That's where the issue is I would bet

**Reply by developers192** [reply]
> The first time a message is sent a `while True` loop is started, and reads a line from a file... then does things based on that. If those ifs do not pass then it will permanently block since those are the only asynchronous things I see in this while loop. That's where the issue is I would bet

Thanks for your response! I added a `if not line: return` and it fixed the issue!
